### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by saliceti**
Please validate the proposed changes:
- Pass only strings (like json dashboards etc) between modules instead of dealing with files
- Explicit individual module variables instead of maps of arbitratry variables
- More explicit variable names (name or monitoring_org_name became monitoring_instance_name)
- Decoupled prometheus and influxdb module in prometheus_all
- Use best practice for [provider versions](https://www.terraform.io/docs/configuration/provider-requirements.html#best-practices-for-provider-versions)
- Removed option of custom grafana grafana.ini. Do we have a use case for that?
- Removed option of custom grafan plugins.txt. Do we have a use case for that?
- Generate data source and dashboard file names based on list index (0.yml, 1.yml...)

I used Teaching vacancies to test it: https://github.com/DFE-Digital/teacher-vacancy-service/blob/terraform-monitoring-0-13-4/terraform/monitoring/resources.tf

### CODE REVIEW

File: terraform/modules/prometheus_all/resources.tf [hunk 1]
```diff
@@ -2,54 +2,50 @@ module paas_prometheus_exporter {
...
 
 module influxdb {
   source = "../influxdb"
-  count  = contains(var.enabled_modules, "prometheus") ? 1 : 0
+  count  = contains(var.enabled_modules, "influxdb") ? 1 : 0
 
-  monitoring_org_name = var.monitoring_org_name
-  monitoring_space_id = data.cloudfoundry_space.monitoring.id
+  monitoring_instance_name = var.monitoring_instance_name
+  space_id                 = data.cloudfoundry_space.monitoring.id
```
Comment (vigneshmsft) [inline terraform/modules/prometheus_all/resources.tf]: why `space_id` and not `monitoring_space_id` ?

File: terraform/modules/redis_prometheus_exporter/future/resources.tf [hunk 2]
```diff
@@ -6,13 +6,13 @@ resource cloudfoundry_route redis_exporter {
 
 resource cloudfoundry_app redis_exporter {
   name         = "redis-exporter-${var.name}"
-  space        =  var.space_id
+  space        = var.space_id
```
Comment (vigneshmsft) [inline terraform/modules/redis_prometheus_exporter/future/resources.tf]: can `future` directory be deleted?
if not this should be `monitor_space_id` ?

File: terraform/modules/postgres_prometheus_exporter/future/resources.tf [hunk 3]
```diff
@@ -6,13 +6,13 @@ resource cloudfoundry_route redis_exporter {
 
 resource cloudfoundry_app redis_exporter {
   name         = "redis-exporter-${var.name}"
-  space        =  var.space_id
+  space        = var.space_id
```
Comment (vigneshmsft) [inline terraform/modules/postgres_prometheus_exporter/future/resources.tf]: same as other comment, can we delete this file?

File: terraform/modules/influxdb/input.tf [hunk 4]
```diff
@@ -1,3 +1,2 @@
-variable monitoring_org_name {}
-
-variable monitoring_space_id {}
+variable space_id {}
```
Comment (vigneshmsft) [inline terraform/modules/influxdb/input.tf]: `monitoring_space_id`, to be same as other resources?

File: terraform/modules/prometheus_all/input.tf [hunk 5]
```diff
@@ -1,36 +1,17 @@
...
-  default = {
-    dashboard_directory  = "../grafana/dashboards"
-    datasource_directory = "../grafana/datasources"
-    configuration_file   = "../grafana/config/grafana.ini"
-  }
-}
+variable grafana_google_client_id { default = "" }
+variable grafana_google_client_secret { default = "" }
+variable grafana_admin_password {}
+variable grafana_json_dashboards {}
```
Comment (vigneshmsft) [inline terraform/modules/prometheus_all/input.tf]:
```suggestion
variable grafana_json_dashboards { default = [] }
```
might be worth having this empty when creating an fresh instance and then creating a dashboard from the UI and exporting as a JSON (to be used in subsequent runs).

Comment (vigneshmsft) [inline terraform/modules/prometheus_all/input.tf]: `variable grafana_extra_datasources { default = [] }`

**Review by vigneshmsft — APPROVED** [review]
nice, have left few comments.

File: terraform/modules/prometheus_all/resources.tf (hunk 1 above)
Comment (saliceti) [reply terraform/modules/prometheus_all/resources.tf]: fixed

File: terraform/modules/redis_prometheus_exporter/future/resources.tf (hunk 2 above)
Comment (saliceti) [reply terraform/modules/redis_prometheus_exporter/future/resources.tf]: As discussed with @sfawcett123 I ended up deleting this exporter. Problem gone

File: terraform/modules/postgres_prometheus_exporter/future/resources.tf (hunk 3 above)
Comment (saliceti) [reply terraform/modules/postgres_prometheus_exporter/future/resources.tf]: As discussed with @sfawcett123 I ended up deleting this exporter. Problem gone

File: terraform/modules/influxdb/input.tf (hunk 4 above)
Comment (saliceti) [reply terraform/modules/influxdb/input.tf]: As discussed with @sfawcett123 I ended up deleting this exporter. Problem gone

File: terraform/modules/prometheus_all/input.tf (hunk 5 above)
Comment (saliceti) [reply terraform/modules/prometheus_all/input.tf]: Good point, thanks

Comment (saliceti) [reply terraform/modules/prometheus_all/input.tf]: Added. Thanks!

**Review by sfawcett123 — APPROVED** [review]
Incorporated changes into GiT and ran through the delivery, worked without failure.
Looks good.

**Comment by saliceti** [comment]
Closed in favour of https://github.com/DFE-Digital/bat-platform-building-blocks/pull/66

**Pull request closed without merging**